opm/simulators/wells/SingleWellState.cpp

```diff
@@ -276,9 +282,27 @@ void SingleWellState::update_targets(const Well& ecl_well, const SummaryState& s
         this->update_injector_targets(ecl_well, st);
 }
 
+bool SingleWellState::operator==(const SingleWellState& rhs) const
+{
+    return this->name == rhs.name &&
```

Comment:
The `serializeOp()` does not serialize all members, but leaves out the `parallel_info`, `pu`, and `perf_data` members. All of these will be set in the (only) constructor, so one is guaranteed to have a valid object after deserialization in the sense of all members being assigned.

However, the `operator==()` is not comparing those three members. So one can have very different SingleWellState objects that compare equal, by for example passing in different `perf_input` arguments. I think we should also compare the three left-out members: `parallel_info` maybe by pointer (not 100% sure about that) and the others by their values.

I think the `pu` and `perf_data` should therefore be added to the serialization.

Reply:
PerfData seems quite straightforward, and also PhaseUsage, and they are held by value here. I understand that parallel_info is trickier.

Reply:
phase usage must be passed through the constructor and if those are not the same across runs we have bigger problems. the perf_data is recreated when a new report step is initialized so it's pointless to serialize it. the parallel well info is a reference so i cannot deserialize it. it is also recreated on the start of report steps so it would be pointless to store it.